app/services/group_message.rb

```diff
@@ -41,6 +41,23 @@ def create
     end
   end
 
+  def delete_previous!
+    posts = Post
+      .joins(topic: { topic_allowed_groups: :group })
+      .where(topic: { posts_count: 1 })
+      .where(topic: { user_id: Discourse.system_user })
+      .where(topic: { archetype: Archetype.private_message })
+      .where(topic: { subtype: TopicSubtype.system_message })
+      .where("posts.created_at > ?", 3.months.ago)
+      .where(topic: { title: I18n.t("system_messages.#{@message_type}.subject_template", message_params) })
+      .where(topic: { topic_allowed_groups: { groups: { name: @group_name } } })
+      .where(raw: I18n.t("system_messages.#{@message_type}.text_body_template", message_params).rstrip)
```

Comment:
```suggestion
    posts = Post
      .joins(topic: { topic_allowed_groups: :group })
      .where(topic: {
        posts_count: 1,
        user_id: Discourse.system_user,
        archetype: Archetype.private_message,
        subtype: TopicSubtype.system_message,
        title: I18n.t("system_messages.#{@message_type}.subject_template", message_params),
        topic_allowed_groups: { 
          groups: { name: @group_name } 
        }
      })
      .where("posts.created_at > ?", 3.months.ago)
      .where(raw: I18n.t("system_messages.#{@message_type}.text_body_template", message_params).rstrip)
```

I kind of feel like this is much clearer as opposed to splitting up the where clause.